server/server.js

```diff
@@ -1122,7 +1164,7 @@ let needSetup = false;
             }
         });
 
-        socket.on("changePassword", async (password, callback) => {
+        socket.on("changePassword", async (userID, password, callback) => {
```

Comment:
I am unhappy that `userID` can be `undefined`
This makes this function quite unintuitive (as in: this could lead to security problems down the line)

Is this nessesary?

Reply:
I'm unhappy too but dont found a better solution.

In the webapp:
- I don't find the `id` of the logged in user in `$root` (which is used to share the app state).
- I need a way to change the password of a user which is not the connected one (the corresponding feature is: an admin user can change the password of another admin user, useful in case of lost password)

In the API, previously, the `changePassword` function use `socket.userID` (aka the logged in user).

At the end, the simplest solution I found is:
- add (kind of optional) `userID` params, which allow to specify the user concerned by the `changePassword` call
- if `userID` is undefined, then we change the password of the logged in user (keeping the existing behavior)

The only way to enhance that would be to store the logged in `userID` in `$root` but that means more changes, not directly related to the feature I try to add... Should I do it anyway? Or perhaps you have a better suggestion?

Reply:
⬆️  @CommanderStorm? 🙃

Reply:
I don't have a good answer, I will have to look into this on the weekend. (that is why you did not get a reply)
Sorry.

Reply:
Just taking a look on changes required to apply my enhancement suggestion: it's very easy! Only 3 lines ☺️ 

Here is the patch:
```diff
diff --git a/server/server.js b/server/server.js
index ea8a207a..3eb19514 100644
--- a/server/server.js
+++ b/server/server.js
@@ -359,6 +359,7 @@ let needSetup = false;
                     callback({
                         ok: true,
                         token: jwt.sign({
+                            userID: user.id,
                             username: data.username,
                         }, server.jwtSecret),
                     });
diff --git a/src/mixins/socket.js b/src/mixins/socket.js
index fabd3440..8551f5f5 100644
--- a/src/mixins/socket.js
+++ b/src/mixins/socket.js
@@ -31,6 +31,7 @@ export default {
                 connectCount: 0,
                 initedSocketIO: false,
             },
+            userID: null,
             username: null,
             remember: (localStorage.remember !== "0"),
             allowLoginDialog: false,        // Allowed to show login dialog, but "loggedIn" have to be true too. This exists because prevent the login dialog show 0.1s in first before the socket server auth-ed.
@@ -374,7 +375,7 @@ export default {
                     this.storage().token = res.token;
                     this.socket.token = res.token;
                     this.loggedIn = true;
-                    this.username = this.getJWTPayload()?.username;
+                    Object.assign(this, this.getJWTPayload());
 
                     // Trigger Chrome Save Password
                     history.pushState({}, "");
```

IMHO it's perfectly normal to have the user id on login, especially on a multi user app. In addition, this PR add a way of listing users (with there id), so I don't see any reason to hide the logged in user id 🤷🏼‍♂️

Reply:
I just add a commit which solve this point, please take a look 😉

Reply:
Resolved